src/main/java/org/openrewrite/dotnet/UpgradeAssistant.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.dotnet;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.*;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class UpgradeAssistant extends UpgradeAssistantRecipe {
+
+    @Option(
+            displayName = "Target framework version",
+            description = "Target framework to which source project should be upgraded.",
+            example = "net9.0")
```

Comment:
I don't know this tool well enough, but would it make sense to limit the values here?
Would also be safer in terms of validation & passing arguments into execution.
```suggestion
            valid = {"net5.0", "net6.0", "net7.0", "net8.0", "net9.0"})
```

Reply:
In parallel I'm wondering if it makes sense to add a declarative recipe that helps discoverability. Assuming there's no need to chain recipes that would look something like this:

`src/main/resources/META-INF/rewrite/dotnet.yml`:
```yaml
---
type: specs.openrewrite.org/v1beta/recipe
name: org.openrewrite.dotnet.MigrateToNet5
recipeList:
  - org.openrewrite.dotnet.UpgradeAssistant:
      targetFramework: net5.0
---
type: specs.openrewrite.org/v1beta/recipe
name: org.openrewrite.dotnet.MigrateToNet6
recipeList:
  - org.openrewrite.dotnet.UpgradeAssistant:
      targetFramework: net6.0
---
type: specs.openrewrite.org/v1beta/recipe
name: org.openrewrite.dotnet.MigrateToNet7
recipeList:
  - org.openrewrite.dotnet.UpgradeAssistant:
      targetFramework: net7.0
---
type: specs.openrewrite.org/v1beta/recipe
name: org.openrewrite.dotnet.MigrateToNet8
recipeList:
  - org.openrewrite.dotnet.UpgradeAssistant:
      targetFramework: net8.0
---
type: specs.openrewrite.org/v1beta/recipe
name: org.openrewrite.dotnet.MigrateToNet9
recipeList:
  - org.openrewrite.dotnet.UpgradeAssistant:
      targetFramework: net9.0
```